apps/system/js/audio_channel_manager.js

```diff
@@ -0,0 +1,267 @@
+/* global BaseModule */
+'use strict';
+
+(function() {
+  var PLAY_CURRENT_AND_NEW_CHANNEL = 0;
+  var PAUSE_CURRENT_CHANNEL = 1;
+  var PAUSE_THEN_RESUME_CURRENT_CHANNEL = 2;
+  var VOLUME_DOWN_CURRENT_CHANNEL = 3;
+  var VOLUME_DOWN_NEW_CHANNEL = 4;
+  var PAUSE_THEN_VIBRATE_CURRENT_CHANNEL = 5;
+  var VIBRATE_NEW_CHANNEL = 6;
+
+  var AudioChannelManager = function() {};
+
+  BaseModule.create(AudioChannelManager, {
+    name: 'AudioChannelManager',
+    DEBUG: false,
+    _currentChannels: null,
+    _pausedChannels: null,
+    _smallerVolumeChannels: null,
+
+    /**
+     * Initial the module.
+     */
+    _start: function() {
+      this._currentChannels = [];
+      this._pausedChannels = [];
+      this._smallerVolumeChannels = [];
+    },
+
+    /**
+     * Add an app into manager to manage its audio channels.
+     * @param {MozBrowser} app An app you want to manage.
+     */
+    register: function(app) {
+      app.allowedAudioChannels.forEach(function(channel) {
+        channel.onactivestatechanged = function(evt) {
+          var channel = evt.target;
+          if (channel.isActive()) {
+            this._handleAudioChannelCompeting(channel);
+          } else {
+            this._resumeLatestPausedChannel();
```

Comment:
Resume or not should be part of the policy. i.e., I don't think we should always resume if there is an inactive audiochannel.